SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialActorResolutionTest/SpatialActorResolutionTest.cpp

```diff
@@ -0,0 +1,58 @@
+﻿#include "SpatialActorResolutionTest.h"
+
+#include "Engine/NetDriver.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+
+ASelfDestroyingActor::ASelfDestroyingActor()
+{
+	bReplicates = true;
+}
+
+bool ASelfDestroyingActor::ReplicateSubobjects(UActorChannel* Channel, FOutBunch* Bunch, FReplicationFlags* RepFlags)
+{
+	TSharedPtr<FDelegateHandle> DelegateHandle = MakeShared<FDelegateHandle>();
+
+	// OnEndFrame decouples us from tick rates and gives no opportunity for the NetDriver to tick in-between.
+	*DelegateHandle = FCoreDelegates::OnEndFrame.AddLambda([this, DelegateHandle]() {
+		Destroy();
+		FCoreDelegates::OnEndFrame.Remove(*DelegateHandle);
+	});
+
+	return Super::ReplicateSubobjects(Channel, Bunch, RepFlags);
+}
+
+ASpatialActorResolutionTest::ASpatialActorResolutionTest()
+{
+	Author = TEXT("Dmitrii <dmitriikozlov@improbable.io>");
+}
+
+/*
+ * This test creates an actor that destroys itself after replicating for the first time, then
+ * waits to check if there are any errors. Created to validate UNR-5752.
+ */
+void ASpatialActorResolutionTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("Spawn the self-destroying actor and save its EntityID"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		TestActor = GetWorld()->SpawnActor<ASelfDestroyingActor>();
+		FinishStep();
+	});
+
+	AddStep(TEXT("Wait until the test actor is destroyed"), FWorkerDefinition::Server(1), nullptr, nullptr, [this](float) {
+		RequireTrue(TestActor.IsStale(), TEXT("The actor became invalid"));
+		FinishStep();
+	});
+
+	AddStep(
+		TEXT("Wait for a predetermined timeout to see there's no errors"), FWorkerDefinition::Server(1), nullptr,
```

Comment:
This is a fuzzy way to get a test result, but alright. Did you check the test 100% fails without your fix?

Reply:
I agree - I couldn't really think of a better check that doesn't involve getting into the ActorSystem's guts, although I hope there is one I'm just ignorant about.
Yes I did - you don't really need this second as the Runtime response usually arrives within a couple ticks.